fix failed to converge within 100 iterations, ruff-0.0.206

command: `ruff --fix $file  # code posted below`

output:
```
weldx/transformations/rotation.py:22:5: D203 1 blank line required before class docstring
weldx/transformations/rotation.py:22:5: D213 Multi-line docstring summary should start at the second line
weldx/transformations/rotation.py:29:9: D213 Multi-line docstring summary should start at the second line
weldx/transformations/rotation.py:39:9: D213 Multi-line docstring summary should start at the second line
weldx/transformations/rotation.py:49:9: D213 Multi-line docstring summary should start at the second line
weldx/transformations/rotation.py:65:9: D213 Multi-line docstring summary should start at the second line
Found 106 error(s) (100 fixed, 6 remaining).

```

pyproject.toml (ruff.tool section):

```toml
[tool.ruff]
exclude = [
    '__init__.py',
    'doc/conf.py',
]
# TODO: should be the following list, but Ruff does not yet impl all of them.
#    W503,W504
#    E203
#    C408
ignore = ['C408','E402']
line-length = 88
select = [
    'B',
    'C',
    'D',
    'E',
    'F',
    'W',
    'UP', # pyupgrade
    'T',  # flake8-print
    #'B950'  # not yet implemented by Ruff.
]

external = ["B950"]
pydocstyle = {convention = "numpy"}
pyupgrade = {keep-runtime-typing = true}
mccabe = {max-complexity = 15}
target-version = "py38"
per-file-ignores = {"**/{tests,tags,asdf}/**" = ["D"]}
```

code:
``` python
"""Contains tools to handle rotations."""
from __future__ import annotations

from typing import TYPE_CHECKING, Union

import pint
from scipy.spatial.transform import Rotation as _Rotation

from weldx.constants import U_
from weldx.constants import WELDX_UNIT_REGISTRY as UREG

if TYPE_CHECKING:  # pragma: no cover
    import numpy.typing as npt

_DEFAULT_LEN_UNIT = UREG.millimeters
_DEFAULT_ANG_UNIT = UREG.rad

ROT_META = "wx_rot"


class WXRotation(_Rotation):
    """Wrapper for creating meta-tagged `~scipy.spatial.transform.Rotation` objects.

    See `scipy.spatial.transform.Rotation` docs for details.
    """

    @classmethod
    def from_quat(cls, quat: npt.ArrayLike) -> "WXRotation":  # noqa
        """Initialize from quaternions.

        See `scipy.spatial.transform.Rotation.from_quat` docs for details.
        """
        rot = super().from_quat(quat)
        setattr(rot, ROT_META, {"constructor": "from_quat"})
        return rot

    @classmethod
    def from_matrix(cls, matrix: npt.ArrayLike) -> "WXRotation":  # noqa
        """Initialize from matrix.

        See `scipy.spatial.transform.Rotation.from_matrix` docs for details.
        """
        rot = super().from_matrix(matrix)
        setattr(rot, ROT_META, {"constructor": "from_matrix"})
        return rot

    @classmethod
    def from_rotvec(cls, rotvec: npt.ArrayLike) -> "WXRotation":  # noqa
        """Initialize from rotation vector.

        See `scipy.spatial.transform.Rotation.from_rotvec` docs for details.
        """
        rot = super().from_rotvec(rotvec)
        setattr(rot, ROT_META, {"constructor": "from_rotvec"})
        return rot

    @classmethod
    @UREG.check(None, None, "[]", None)
    def from_euler(
        cls,
        seq: str,
        angles: Union[pint.Quantity, npt.ArrayLike],
        degrees: bool = False,
    ) -> "WXRotation":  # noqa
        """Initialize from euler angles.

        See `scipy.spatial.transform.Rotation.from_euler` docs for details.
        """
        if isinstance(angles, pint.Quantity):
            if angles.u == U_(""):
                angles = angles.to("rad")
            degrees = "rad" not in str(angles.u)
            if degrees:
                angles = angles.to("degree")
            else:
                angles = angles.to("rad")
            angles = angles.m

        rot = super().from_euler(seq=seq, angles=angles, degrees=degrees)
        setattr(
            rot,
            ROT_META,
            {"constructor": "from_euler", "seq": seq, "degrees": degrees},
        )
        return rot
```

--- a/src/registry.rs
+++ b/src/registry.rs
@@ -3798,6 +3798,14 @@ impl Check {
     }
 }
 
+/// Pairs of checks that shouldn't be enabled together.
+pub const INCOMPATIBLE_CODES: &[(CheckCode, CheckCode, &str)] = &[(
+    CheckCode::D203,
+    CheckCode::D211,
+    "`D203` (OneBlankLineBeforeClass) and `D211` (NoBlankLinesBeforeClass) are incompatible. \
+     Consider adding `D203` to `ignore`.",
+)];
+
 /// A hash map from deprecated `CheckCodePrefix` to latest `CheckCodePrefix`.
 pub static PREFIX_REDIRECTS: Lazy<FxHashMap<&'static str, CheckCodePrefix>> = Lazy::new(|| {
     FxHashMap::from_iter([

--- a/src/settings/mod.rs
+++ b/src/settings/mod.rs
@@ -6,6 +6,7 @@ use std::hash::{Hash, Hasher};
 use std::path::{Path, PathBuf};
 
 use anyhow::{anyhow, Result};
+use colored::Colorize;
 use globset::{Glob, GlobMatcher, GlobSet};
 use itertools::Itertools;
 use once_cell::sync::Lazy;
@@ -14,7 +15,7 @@ use regex::Regex;
 use rustc_hash::FxHashSet;
 
 use crate::cache::cache_dir;
-use crate::registry::CheckCode;
+use crate::registry::{CheckCode, INCOMPATIBLE_CODES};
 use crate::registry_gen::{CheckCodePrefix, SuffixLength, CATEGORIES};
 use crate::settings::configuration::Configuration;
 use crate::settings::types::{
@@ -23,7 +24,7 @@ use crate::settings::types::{
 use crate::{
     flake8_annotations, flake8_bugbear, flake8_errmsg, flake8_import_conventions,
     flake8_pytest_style, flake8_quotes, flake8_tidy_imports, flake8_unused_arguments, isort,
-    mccabe, pep8_naming, pydocstyle, pyupgrade,
+    mccabe, one_time_warning, pep8_naming, pydocstyle, pyupgrade,
 };
 
 pub mod configuration;
@@ -32,7 +33,6 @@ pub mod options;
 pub mod options_base;
 pub mod pyproject;
 pub mod types;
-
 const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 #[derive(Debug)]
@@ -113,7 +113,7 @@ impl Settings {
             dummy_variable_rgx: config
                 .dummy_variable_rgx
                 .unwrap_or_else(|| DEFAULT_DUMMY_VARIABLE_RGX.clone()),
-            enabled: resolve_codes(
+            enabled: validate_enabled(resolve_codes(
                 config
                     .pydocstyle
                     .as_ref()
@@ -134,7 +134,7 @@ impl Settings {
                         .zip(config.extend_ignore.iter())
                         .map(|(select, ignore)| CheckCodeSpec { select, ignore }),
                 ),
-            ),
+            )),
             exclude: resolve_globset(config.exclude.unwrap_or_else(|| DEFAULT_EXCLUDE.clone()))?,
             extend_exclude: resolve_globset(config.extend_exclude)?,
             external: FxHashSet::from_iter(config.external.unwrap_or_default()),
@@ -420,6 +420,21 @@ fn resolve_codes<'a>(
         }
     }
     codes
+}
+
+/// Warn if the set of enabled codes contains any incompatibilities.
+fn validate_enabled(enabled: FxHashSet<CheckCode>) -> FxHashSet<CheckCode> {
+    for (a, b, message) in INCOMPATIBLE_CODES {
+        if enabled.contains(a) && enabled.contains(b) {
+            one_time_warning!(
+                "{}{} {}",
+                "warning".yellow().bold(),
+                ":".bold(),
+                message.bold()
+            );
+        }
+    }
+    enabled
 }
 
 #[cfg(test)]

